pkg/codegen/externalref.go

```diff
@@ -0,0 +1,80 @@
+package codegen
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ensureExternalRefsInRequestBodyDefinitions ensures that when an externalRef (`$ref` that points to a file that isn't the current spec) is encountered, we make sure we update our underlying `RefType` to make sure that we point to that type.
+// This only happens if we have a non-empty `ref` passed in, and that `ref` isn't pointing to something in our file
+// NOTE that the pointer here allows us to pass in a reference and edit in-place
+func ensureExternalRefsInRequestBodyDefinitions(defs *[]RequestBodyDefinition, ref string) {
```

Comment:
I think it would be easier to understand if the param was `defs []*RequestBodyDefinition`, since the underlying def could be changed directly in that case, and you wouldn't need the `(*defs)[i] = rbd`. Or is there something I'm missing?

Reply:
That's a fair shout.

I went for this solution, as I could just pass it in from `operations.go`:

```go
			ensureExternalRefsInRequestBodyDefinitions(&bodyDefinitions, pathItem.Ref)
```

Whereas if I went with that solution, I'd need to create a new slice that had a pointer to each value within `bodyDefinitions` :thinking: